instrumentation/opentelemetry_bandit/test/bandit/opentelemetry_bandit_test.exs

```diff
@@ -1,243 +1,464 @@
+                        name: :GET,
+                        kind: :server,
+                        parent_span_id: 13_235_353_014_750_950_193
+                      )}
+
+      refute_receive {:span,
+                      span(
+                        name: :GET,
+                        kind: :server,
+                        parent_span_id: :undefined
                       )}
 
-      assert %{
-               "http.url": _,
-               "http.method": "GET",
-               "http.status_code": 500,
-               "http.response_content_length": 10,
-               "net.transport": :"IP.TCP"
-             } = :otel_attributes.map(attributes)
+      System.delete_env("PUBLIC")
     end
 
-    test "span when request_target is empty" do
-      :telemetry.execute(
-        [:bandit, :request, :stop],
-        %{duration: 444, resp_body_bytes: 10},
-        %{
-          conn: %{status: 500, method: "GET", request_target: nil},
-          error: "Internal Server Error"
-        }
+    test "with all opt-ins" do
+      OpentelemetryBandit.setup(
+        opt_in_attrs: [
+          {ClientAttributes.client_port(), true},
+          {HTTPAttributes.http_request_body_size(), true},
+          {HTTPAttributes.http_response_body_size(), true},
+          {NetworkAttributes.network_local_address(), true},
+          {NetworkAttributes.network_local_port(), true},
+          {NetworkAttributes.network_transport(), true}
+        ],
+        request_headers: ["test-header"],
+        response_headers: ["content-type"]
+      )
+
+      port = start_server()
+
+      Req.get!("http://localhost:#{port}/with_body",
+        headers: %{"test-header" => "request header"}
       )
 
       assert_receive {:span,
                       span(
-                        name: "HTTP GET",
-                        kind: :error,
-                        status: {:status, :error, "Internal Server Error"},
-                        attributes: attributes
+                        name: :GET,
+                        attributes: span_attrs
                       )}
 
-      assert %{
-               "http.url": _,
-               "http.method": "GET",
-               "http.status_code": 500,
-               "http.response_content_length": 10,
-               "net.transport": :"IP.TCP"
-             } = :otel_attributes.map(attributes)
+      attrs = :otel_attributes.map(span_attrs)
+
+      expected_attrs = [
+        {ClientAttributes.client_address(), "127.0.0.1"},
+        {HTTPAttributes.http_request_body_size(), 0},
+        {HTTPAttributes.http_request_method(), :GET},
+        {HTTPAttributes.http_response_body_size(), 29},
+        {HTTPAttributes.http_response_status_code(), 200},
+        {String.to_atom("#{HTTPAttributes.http_request_header()}.test-header"),
+         ["request header"]},
+        {String.to_atom("#{HTTPAttributes.http_response_header()}.content-type"),
+         ["application/json; charset=utf-8"]},
```

Comment:
It looks like `:otel_http` is building atoms based on request/response headers? That seems kinda dangerous to me.

Reply:
It can only create whatever you told it to. There's no option to capture all headers so it is bounded